File: python/examples/streaming.py
"""
Streaming example for Claude Code SDK
"""

import os
import asyncio
from claude_code_sdk import ClaudeCode

# Initialize the client
claude = ClaudeCode(options={
    "api_key": os.environ.get("ANTHROPIC_API_KEY")
})

# OpenAI-style streaming
async def stream_code():
    stream = claude.chat["completions"].create_stream({
        "model": "claude-code",
        "messages": [
            {"role": "user", "content": "Create a Python class for a login form"}
        ],
        "stream": True
    })
    
    async for chunk in stream:
        if "choices" in chunk and chunk["choices"] and "delta" in chunk["choices"][0]:
            delta = chunk["choices"][0]["delta"]
            if "content" in delta and delta["content"]:
                print(delta["content"], end="", flush=True)

# Anthropic-style streaming
async def stream_code_anthropic():
    stream = claude.messages.create_stream({
        "model": "claude-code",
        "messages": [
            {
                "role": "user",
                "content": [
                    {"type": "text", "text": "Create a Python class for a login form"}
                ]
            }
        ],
        "stream": True
    })
    
    async for chunk in stream:
        if chunk.get("type") == "content_block_delta" and "delta" in chunk:
            delta = chunk["delta"]
            if "text" in delta:
                print(delta["text"], end="", flush=True)

# Session streaming
async def stream_session():
    # Start a session
    session = claude.sessions.create({
        "messages": [
            {"role": "user", "content": "Let's create a Python project"}
        ]
    })
    
    # Continue the session with streaming
    stream = session.continue_stream({
        "messages": [
            {"role": "user", "content": "Now add a database connection"}
        ]
    })
    
    async for chunk in stream:
        if "choices" in chunk and chunk["choices"] and "delta" in chunk["choices"][0]:
            delta = chunk["choices"][0]["delta"]
            if "content" in delta and delta["content"]:
                print(delta["content"], end="", flush=True)

if __name__ == "__main__":
    asyncio.run(stream_code())

File: typescript/src/examples/streaming.ts
import { ClaudeCode } from '../index'

// Example of streaming responses with Claude Code SDK

async function streamExample(): Promise<void> {
  // Create a client
  const claude = new ClaudeCode({
    apiKey: process.env.ANTHROPIC_API_KEY,
  })

  try {
    console.log('Streaming with OpenAI-style API:')
    console.log('--------------------------------')
    
    // OpenAI-style streaming
    const openaiStream = await claude.chat.completions.createStream({
      model: 'claude-code',
      messages: [
        { role: 'user', content: 'Write a React component for a to-do list with TypeScript' }
      ]
    })

    console.log('Response:')
    for await (const chunk of openaiStream) {
      if (chunk.choices && chunk.choices[0]?.delta?.content) {
        process.stdout.write(chunk.choices[0].delta.content)
      }
    }
    
    console.log('\n\n')
    console.log('Streaming with Anthropic-style API:')
    console.log('---------------------------------')
    
    // Anthropic-style streaming
    const anthropicStream = await claude.messages.createStream({
      model: 'claude-code',
      messages: [
        { 
          role: 'user', 
          content: [{ 
            type: 'text', 
            text: 'Write a TypeScript utility for handling API errors' 
          }] 
        }
      ]
    })

    console.log('Response:')
    for await (const chunk of anthropicStream) {
      if (chunk.type === 'content_block_delta' && chunk.delta?.text) {
        process.stdout.write(chunk.delta.text)
      }
    }
    
    console.log('\n')
    
  } catch (error) {
    console.error('Error:', error)
  }
}

// Execute the example if this file is run directly
if (require.main === module) {
  streamExample().catch(console.error)
}

export default streamExample